src/components/ApiHistoryTable.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import { Details } from '@docusaurus/theme-common/Details';
+import {
+  ApiHistory,
+  PrReleaseVersions,
+  PrReleaseVersionsContainer,
+} from '../transformers/api-history';
+import ReactMarkdown from 'react-markdown';
+
+import styles from './ApiHistoryTable.module.scss';
+
+enum Change {
+  ADDED = 'API ADDED',
+  CHANGED = 'API CHANGED',
+  DEPRECATED = 'API DEPRECATED',
+}
+
+interface ApiHistoryTableProps {
+  apiHistoryJson: string;
+  prReleaseVersionsJson: string;
+}
+
+function generateTableRow(
+  prReleaseVersions: PrReleaseVersions | undefined,
+  type: Change,
+  change:
+    | NonNullable<ApiHistory['added']>[0]
+    | NonNullable<ApiHistory['deprecated']>[0]
+    | NonNullable<ApiHistory['changes']>[0]
+) {
+  const prNumber = Number(change['pr-url'].split('/').at(-1));
```

Comment:
Maybe we should add a `prNumber` parameter to this function?

Reply:
I had it that way before ¯\\_(ツ)_/¯

```patch
    ...(apiHistory.changes?.map((change) => {
      const prNumber = Number(change['pr-url'].split('/').at(-1));
      return generateTableRow(
+		prNumber 
        prReleaseVersions[prNumber],
        Change.CHANGED,
        change
      );
    }) ?? []),
```

I personally prefer the way it is currently.

Reply:
Because I found a lot of code like `Number(added['pr-url'].split('/').at(-1))` appearing, I raised this question to see if there's a better way to optimize this code.

Of course, over-encapsulation and optimization are sometimes not a good idea. So, this is just a personal opinion :)

> BTW, regarding code style, I'm just giving my opinion. The final decision still lies with the person doing this.